sklearn/utils/validation.py

```diff
@@ -2167,18 +2174,24 @@ def _check_sample_weight(
     sample_weight : ndarray of shape (n_samples,)
         Validated sample weight. It is guaranteed to be "C" contiguous.
     """
+    xp, _, device = get_namespace_and_device(sample_weight, X)
```

Comment:
Isn't this function already being modified in another array api PR?

Reply:
Yes! In the roc_auc one. I realised when I reviewed it. I think that should go in first, as it does include tests, and I will just merge main.